File: 80-remove-duplicates-from-sorted-array-ii/80-remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        map = dict()
        cur_index = 1
        map[nums[0]]=1
        
        for i in range(1,len(nums)):
            if nums[i]!=nums[i-1]:
                nums[cur_index]=nums[i]
                cur_index+=1
                map[nums[i]]=1
            elif map[nums[i]]<2:
                nums[cur_index]=nums[i]
                cur_index+=1
                map[nums[i]]+=1
        print(nums)               
        return cur_index

File: 80-remove-duplicates-from-sorted-array-ii/80-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        
        Map<Integer,Integer> map=new HashMap<>();
        int cur_index=1;
        map.put(nums[0],1);
        
        for(int i=1;i<nums.length;i++)
        {
            if(nums[i]!=nums[i-1])
            {
                nums[cur_index++]=nums[i];
                map.put(nums[i],1);
            }
            else if(map.get(nums[i])<2)
            {
                nums[cur_index++]=nums[i];
                map.put(nums[i],map.get(nums[i])+1);
            }
        }
        return cur_index;
    }
}